app/common/adapter/changesStream/CnpmjsorgChangesStream.ts

```diff
@@ -0,0 +1,69 @@
+import { ContextProto } from '@eggjs/tegg';
+import { Readable } from 'node:stream';
+import { RegistryType } from 'app/common/enum/Registry';
+import { Registry } from 'app/core/entity/Registry';
+import { E500 } from 'egg-errors';
+import { AbstractChangeStream, ChangesStreamChange, RegistryChangesStream } from './AbstractChangesStream';
+
+const MAX_LIMIT = 10000;
+
+@ContextProto()
+@RegistryChangesStream(RegistryType.Cnpmjsorg)
+export class CnpmjsorgChangesStream extends AbstractChangeStream {
+
+  // cnpmjsorg 未实现 update_seq 字段
+  // 默认返回当前时间戳字符串
+  async getInitialSince(registry: Registry): Promise<string> {
+    const since = String((new Date()).getTime() - 1);
+    this.logger.warn(`[CnpmjsorgChangesStream.getInitialSince] since: 1, skip query ${registry.changeStream}`);
+    return since;
+  }
+
+  private async tryFetch(registry: Registry, since: string, limit = 1000) {
+    if (limit > MAX_LIMIT) {
+      throw new E500(`limit too large, current since: ${since}, limit: ${limit}`);
+    }
+    const db = `${registry.changeStream}?since=${since}&limit=${limit}`;
```

Comment:
这类 querystring 操作，可以直接用原生 API，不容易出错。
```javascript
const dbUrl = new URL(registry.changeStream);
dbUrl.searchParams.set('since', since);
dbUrl.searchParams.set('limit', limit);

const db = dbUrl.toString();
```

Reply:
加个 getChangesStream 方法一起处理